packages/contracts-bedrock/src/L2/OptimismSuperchainERC20.sol

```diff
@@ -0,0 +1,187 @@
+    }
+
+    /// @notice Sends tokens to some target address on another chain.
+    /// @param _to      Address to send tokens to.
+    /// @param _amount  Amount of tokens to send.
+    /// @param _chainId Chain ID of the destination chain.
+    function sendERC20(address _to, uint256 _amount, uint256 _chainId) external {
+        if (_to == address(0)) revert ZeroAddress();
+
+        _burn(msg.sender, _amount);
+
+        bytes memory _message = abi.encodeCall(this.relayERC20, (msg.sender, _to, _amount));
+        IL2ToL2CrossDomainMessenger(MESSENGER).sendMessage(_chainId, address(this), _message);
+
+        emit SendERC20(msg.sender, _to, _amount, _chainId);
+    }
+
+    /// @notice Relays tokens received from another chain.
+    /// @param _from   Address of the msg.sender of sendERC20 on the source chain.
+    /// @param _to     Address to relay tokens to.
+    /// @param _amount Amount of tokens to relay.
+    function relayERC20(address _from, address _to, uint256 _amount) external {
+        if (_to == address(0)) revert ZeroAddress();
+
+        if (msg.sender != MESSENGER) revert CallerNotL2ToL2CrossDomainMessenger();
+
+        if (IL2ToL2CrossDomainMessenger(MESSENGER).crossDomainMessageSender() != address(this)) {
+            revert InvalidCrossDomainSender();
+        }
+        uint256 _source = IL2ToL2CrossDomainMessenger(MESSENGER).crossDomainMessageSource();
+
+        _mint(_to, _amount);
+
+        emit RelayERC20(_from, _to, _amount, _source);
+    }
+
+    /// @notice Returns the address of the corresponding version of this token on the remote chain.
+    function remoteToken() public view override returns (address) {
+        return _getMetadataStorage().remoteToken;
+    }
+
+    /// @notice Returns the name of the token.
+    function name() public view virtual override returns (string memory) {
+        return _getMetadataStorage().name;
+    }
+
+    /// @notice Returns the symbol of the token.
+    function symbol() public view virtual override returns (string memory) {
+        return _getMetadataStorage().symbol;
+    }
+
+    /// @notice Returns the number of decimals used to get its user representation.
+    /// For example, if `decimals` equals `2`, a balance of `505` tokens should
+    /// be displayed to a user as `5.05` (`505 / 10 ** 2`).
+    /// NOTE: This information is only used for _display_ purposes: it in
+    /// no way affects any of the arithmetic of the contract, including
+    /// {IERC20-balanceOf} and {IERC20-transfer}.
+    function decimals() public view override returns (uint8) {
+        return _getMetadataStorage().decimals;
+    }
+
+    /// @notice ERC165 interface check function.
+    /// @param _interfaceId Interface ID to check.
+    /// @return Whether or not the interface is supported by this contract.
+    function supportsInterface(bytes4 _interfaceId) external pure virtual returns (bool) {
+        return _interfaceId == type(IOptimismSuperchainERC20Extension).interfaceId;
```

Comment:
I believe this should also call the parent method so it doesn't return false on calls to e.g. `type(ERC20).interfaceId`